rock/python.rb

```diff
@@ -212,30 +212,36 @@ def self.deactivate_python(ws: Autoproj.workspace)
         ws.config.reset('python_version')
     end
 
-    # Allow to update the PYTHONPATH for package, tries to guess the python
-    # binary from Autobuild.programs['python'] and system's default setting
+    # Allow to update the PYTHONPATH for package if autoproj configuration
+    # USE_PYTHON is set to true.
+    # Then tries to guess the python binary from Autobuild.programs['python']
+    # and system's default setting
     # @param [Autobuild::Package] pkg
     # @param [Autoproj.workspace] ws Autoproj workspace
     # @param [String] bin Path to a custom python version
     # @param [String] version version constraint for python executable
+    # @return tuple of [executable, version, site-packages path] if set,
+    #    otherwise [nil,nil,nil]
     def self.activate_python_path(pkg,
                              ws: Autoproj.workspace,
                              bin: nil,
                              version: nil)
-        bin, version = resolve_python(ws: ws, bin: bin, version: version)
-        path = ws.env.add_path 'PYTHONPATH',
-                   File.join(pkg.prefix, "lib",
-                             "python#{version}","site-packages")
-        [bin, version, path]
+        if ws.config.has_value_for?('USE_PYTHON') and ws.config.get('USE_PYTHON')
+            bin, version = resolve_python(ws: ws, bin: bin, version: version)
+            path = ws.env.add_path 'PYTHONPATH',
+                       File.join(pkg.prefix, "lib",
+                                 "python#{version}","site-packages")
+            return [bin, version, path]
+        end
+        [nil,nil,nil]
```

Comment:
Why not just `nil` ?

Also, would be easier on the reader to return nil first and then go through the rest of the code

~~~
return unless ws.config.get('USE_PYTHON', false)

# Do the rest
~~~